src/Common/src/System/IO/PathInternal.Unix.cs

```diff
@@ -48,5 +51,44 @@ internal static bool IsDirectoryTooLong(string fullPath)
         {
             return fullPath.Length >= Interop.Sys.MaxPath;
         }
+
+        /// <summary>
+        /// Normalize separators in the given path. Compresses forward slash runs.
+        /// </summary>
+        internal static string NormalizeDirectorySeparators(string path)
+        {
+            if (string.IsNullOrWhiteSpace(path)) return path;
+
+            // Make a pass to see if we need to normalize so we can potentially skip allocating
+            bool normalized = true;
+
+            for (int i = 0; i < path.Length; i++)
+            {
+                if (IsDirectorySeparator(path[i])
+                    && (i < path.Length - 1 && IsDirectorySeparator(path[i + 1])))
+                {
+                    normalized = false;
+                    break;
+                }
+            }
+
+            if (normalized) return path;
+
+            StringBuilder builder = new StringBuilder(path.Length);
+
+            for (int i = 0; i < path.Length; i++)
+            {
+                char current = path[i];
+
+                // Skip if we have another separator following
+                if (IsDirectorySeparator(path[i])
```

Comment:
Nit: could use current here, otherwise there's not really value in having the local